repo/repo.go

```diff
@@ -98,16 +98,45 @@ func ListRepositories(workspace *rule.File) (repos []*rule.Rule, repoFileMap map
 			if err != nil {
 				return nil, nil, err
 			}
+			leveled := strings.HasPrefix(f, "+")
+			f = strings.TrimPrefix(f, "+")
 			f = filepath.Join(filepath.Dir(workspace.Path), filepath.Clean(f))
 			macroFile, err := rule.LoadMacroFile(f, "", defName)
 			if err != nil {
 				return nil, nil, err
 			}
 			for _, repo := range macroFile.Rules {
-				if name := repo.Name(); name != "" {
-					repos = append(repos, repo)
-					repoFileMap[name] = macroFile
-					repoIndexMap[name] = len(repos) - 1
+				if kind := repo.Kind(); kind == "go_repository" || !leveled {
+					if name := repo.Name(); name != "" {
+						repos = append(repos, repo)
+						repoFileMap[name] = macroFile
+						repoIndexMap[name] = len(repos) - 1
+					}
+				} else {
+					// This block will only be entered if leveled==true
+					var callFile string
+					var defName string
+					for _, l := range macroFile.Loads {
+						if l.Has(kind) {
+							callFile = filepath.Join(filepath.Dir(workspace.Path), filepath.Clean(l.Name()))
+							defName = l.Get(kind)
+							break
+						}
+					}
+					if len(callFile) == 0 {
+						continue
```

Comment:
Add a `TODO` if  you don't want to handle in this PR the situation where one macro calls another macro in the same bzl file, hence no matching `load` statement